### PR TITLE
Sync with http-factory

### PR DESCRIPTION
**Pull request by shadowhand**
Version 0.3.0 introduced changes in server request factory.

### CODE REVIEW

**Comment by shadowhand** [comment]
@Nyholm how about this one?

File: src/ServerRequestFactory.php [hunk 1]
```diff
@@ -18,15 +23,13 @@ public function createServerRequest(array $server, $method = null, $uri = null)
         }
 
         // TODO: find a MUCH better way
```
Comment (Nyholm) [inline src/ServerRequestFactory.php]: Is there a better way? Maybe explain **why** we do this instead.

Comment (shadowhand) [inline src/ServerRequestFactory.php]: I didn't write this line. :) [blame](https://github.com/http-interop/http-factory-guzzle/blame/829e40990675a543f32a9d77b0e1163362b4943c/src/ServerRequestFactory.php#L25)

Comment (Nyholm) [inline src/ServerRequestFactory.php]: =)

**Pull request merged**